paddle/fluid/operators/softmax_with_cross_entropy_op.cu

```diff
@@ -53,8 +55,157 @@ __global__ void SoftCrossEntropyGradientKernel(T* logit_grad,
     logit_grad[ids] = loss_grad[row_ids] * (logit_grad[ids] - labels[ids]);
   }
 }
+
 }  // namespace
 
+__device__ __forceinline__ float real_exp(float x) { return expf(x); }
+__device__ __forceinline__ double real_exp(double x) { return exp(x); }
+__device__ __forceinline__ float real_log(float x) {
+  return math::TolerableValue<float>()(logf(x));
+}
+__device__ __forceinline__ double real_log(double x) {
+  return math::TolerableValue<double>()(log(x));
+}
+
+// There are 3 kinds of reduce algorithm:
+// BLOCK_REDUCE_RAKING_COMMUTATIVE_ONLY
+// BLOCK_REDUCE_RAKING
+// BLOCK_REDUCE_WARP_REDUCTIONS (default)
+template <typename T, int BlockSize>
+using BlockReduce =
+    cub::BlockReduce<T, BlockSize /*, cub::BLOCK_REDUCE_WARP_REDUCTIONS*/>;
+
+template <typename T, int BlockSize>
+using BlockReduceTempStorage = typename BlockReduce<T, BlockSize>::TempStorage;
```

Comment:
Since it is a temporary storage, could we just set BlockSize to MaxBlockSize? It will reduce the binary size.